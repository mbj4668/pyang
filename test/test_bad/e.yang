module e {
    namespace "urn:e";
    prefix e;

    import f { // note that prefix f is used in must expression
        prefix f;
    }

    typedef arnex {
        type int32;
    }

    grouping xxx {
        typedef arne { // warning: unused typedef
            type int32;
        }
        grouping x { // warning: unused grouping
            leaf f {
                type int32;
            }
        }
        leaf z4 {
            type int32;
        }
    }

  container y {
    must "/f:gg[a='f:s']";

    leaf x {
      type int32;
    }
  }

  augment "/e:y/e:z" {
    leaf g {
      type int32;
    }
  }

  augment "/e:y/e:z/e:q" {
    leaf g {
      type int32;
    }
  }

  augment "/e:y" {
    container z {
    }
  }


    augment "/e:y/e:z/e:q/e:w" {
      leaf f {
          type int32;
      }
      container s {
          config false;
      }
    }

    augment "/e:y/e:z/e:q/e:w/e:s" {
        leaf f {
            config true;
            type string;
        }
    }

    augment "/e:y/e:z/e:q/e:w/e:e" { // error: e is a leaf
      leaf f {
          type int32;
      }
  }


    augment "/e:y/e:z/e:q/e:g" { // error: g is a leaf
      leaf f {
          type int32;
      }
    }

    augment "/e:y/e:z/e:q/e:g/e:r" { // error: f is a leaf; r not found
      leaf f {
          type int32;
      }
    }

  augment "/e:y/e:z" {
    container q {
        container w {
            leaf e {
                type int32;
            }
        }
    }
  }

  leaf xx {
    type string;
    must "/a:bar";  // error: prefix a not defined
  }

}
