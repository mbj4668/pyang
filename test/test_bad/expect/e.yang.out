e.yang:14: warning: UNUSED_TYPEDEF
e.yang:17: warning: UNUSED_GROUPING
e.yang:28: warning: XPATH_NODE_NOT_FOUND1
e.yang:64: error: INVALID_CONFIG
e.yang:69: error: BAD_NODE_IN_AUGMENT
e.yang:76: error: BAD_NODE_IN_AUGMENT
e.yang:82: error: NODE_NOT_FOUND
e.yang:100: error: PREFIX_NOT_DEFINED
f.yang:11: error: NEED_KEY
f.yang:25: error: INVALID_CONFIG
f.yang:36: error: BAD_TYPE_IN_KEY
f.yang:38: error: BAD_UNIQUE_PART
